src/eei.cpp

```diff
@@ -550,6 +597,9 @@ namespace HeraVM {
       if (call_result.release)
         call_result.release(&call_result);
 
+      /* Return unspent gas */
+      result.gasLeft += call_result.gas_left;
```

Comment:
For this to work, the entire passed gas must be used via `takeGas` before doing an EVM-C call.

Reply:
This is done by just performing:
```takeGas(gas);```
before the call execution, correct?

Reply:
Yep